### ISSUE TITLE
Update TMs with SDLXLIFF files

### ISSUE THREAD

**Issue by newtriddell** [EvzenP/STraSAK#1]
**New feature**:
There're two good features about STraSAK: 
1) Create new file based translation memories
2) Export the Trados Studio translation memories to TMX files

Is there something that can be added in the middle to update the created translation memories with their respective SDLXLIFF files from the main project? This avoids using third party apps like SDLXliff2Tmx (then exporting TMs to TMXs can do the trick).

**Comment by EvzenP** [comment]
To update main TMs of a project, use the Update-MainTMs command. That should do what you want.

**Comment by EvzenP** [comment]
BTW, the Sdlxliff2Tmx tool does something else than update main TMs of a project. It's rather an equivalent of creating new TM, then importing SDLXLIFFs into it and then exprting the TM to TMX.

Still, it might not be exactly the same as Sdlxliff2Tmx, depending on the import options.

**Comment by newtriddell** [comment]
> It's rather an equivalent of creating new TM, then importing SDLXLIFFs into it and then exprting the TM to TMX.

Exactly, that's what I am looking for. E.g for a multiple language project (or a single language one) I need to create new TMs (for each language), update TMs with the SDLXLIFF files from that project and export them to TMXs. Is that something the script can support? Thanks.

**Comment by EvzenP** [comment]
If you are talking about **project**, then you should **work with project** in the first place.
I.e. you should really create a **project** (which uses you TMs you created beforehand), so that you can then use the `Update-MainTMs` command.
What you want, does not make much sense to me... if you have a **project**, you should have the TMs already created and defined in the project. And if you do, all you need to do is to update the main TMs, that's it. Creating some extra TMs and importing the project SDLXLIFFs to them does not make sense... Or I am missing some fundamental piece of information you have not revealed.

I'm not sure if you realize fully that STraSAK contains just **modules** containing **functions**. It's NOT a script which does something.
Creating a script which uses individual; functions in a flow fitting your needs is on you.
You can find some examples of batch scripts in the Readme.

**Comment by newtriddell** [comment]
Project coordinator here, not a programer, - picked the wrong wording :-). In my day-to-day job I am asked to deliver: 1) target files 2) TMs and TMXs with translation units ONLY from a project (I know, sounds crazy). `Update-MainTMs` command doesn't work for me in such scenarios. That's why I'm trying to figure out a way to: create new TMs > update them with SDLXLIFF files > export TMs to TMXs.

Creating the empty TMs first and use `Update-MainTMs` command would do the trick, but I noticed that I can't create multiple TMs in one go e.g the below command line won't work:

`Call "C:\My scripts\TS2017.cmd" New-FileBasedTM -Name "Translation Memory name" -TMLoc "D:\Project\TMs" -SourceLanguage "en-US" -TargetLanguage "de-DE,fi-FI"`

It throws the below error:

```
New-Object : Exception calling ".ctor" with "8" argument(s): "Value cannot be null.
Parameter name: targetLanguage"
At C:\Users\newtriddell\Documents\WindowsPowerShell\Modules\TMHelper\TMHelper.psm1:107 char:9
+ ...       $TM = New-Object Sdl.LanguagePlatform.TranslationMemoryApi.File ...
+                 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : InvalidOperation: (:) [New-Object], MethodInvocationException
    + FullyQualifiedErrorId : ConstructorInvokedThrowException,Microsoft.PowerShell.Commands.NewObjectCommand
```
Maybe I'm trying to automate too much. Reading through the API but I am far away from calling myself a rookie. Maybe this is something to consider in the future, - a version of STraSAK to support the creation of multiple TMs in one go. Thanks!

**Comment by EvzenP** [comment]
> but I noticed that I can't create multiple TMs in one go e.g the below command line won't work

Ahh, now I understand! A typical [XY problem](http://xyproblem.info/) ;-)

Yes, the error is caused by using a list of target languages instead of a single language.
The `New-FileBasedTM` supports only **single** language as `TargetLanguage`, i.e. you need a self-made  loop to create multiple TMs. Something like this:

```batchfile
set "SourceLang=en-US"
set "TargetLangs=de-DE fi-FI fr-FR"
for %%L in (%TargetLangs%) do (
  call "C:\My scripts\TS2017.cmd" New-FileBasedTM -Name "Translation Memory name %SourceLang%_%%L" -TMLoc "D:\Project\TMs" -SourceLanguage "%SourceLang%" -TargetLanguage "%%L"
)
```

> Maybe this is something to consider in the future, - a version of STraSAK to support the creation of multiple TMs in one go.

Yup, very logical request! Don't really know why didn't I do it right away... it's been a long time ;-)

**Comment by EvzenP** [comment]
I just released version 1.5 which adds support for creation of multiple TMs by `New-FileBasedTM`.
It would be great if you can test it and let me know how it works.

**Comment by newtriddell** [comment]
Just tested and it works like charm. Many thanks for releasing this update. :-) Closing.
